RecipeMatch/View Controllers/StreamViewController.m

```diff
@@ -11,41 +11,61 @@
 #import "SceneDelegate.h"
 #import "DraggableViewBackground.h"
 
+@interface StreamViewController()
+@property (nonatomic, strong) NSString *preferences;
+@property (nonatomic, strong) DraggableViewBackground *draggableBackground;
+@end
+
 @implementation StreamViewController
 
 - (void)viewDidLoad {
     [super viewDidLoad];
     // Do any additional setup after loading the view.
+
+    // add centered logo
+    UIImageView* imageView = [[UIImageView alloc] initWithImage:[UIImage imageNamed:@"logo"]];
+    imageView.contentMode = UIViewContentModeScaleAspectFit;
+
+    UIView* titleView = [[UIView alloc] initWithFrame:CGRectMake(0, 0, 100, 60)];
+    imageView.frame = titleView.bounds;
+    [titleView addSubview:imageView];
+
+    self.navigationItem.titleView = titleView;
     
-    DraggableViewBackground *draggableBackground = [[DraggableViewBackground alloc]initWithFrame:self.view.frame];
-    [self.view addSubview:draggableBackground];
-    
+    UIActivityIndicatorView *spinner = [[UIActivityIndicatorView alloc] initWithActivityIndicatorStyle:UIActivityIndicatorViewStyleMedium];
+    spinner.center = CGPointMake(self.view.center.x, self.view.center.y);
+    spinner.tag = 12;
+    [self.view addSubview:spinner];
+    [spinner startAnimating];
     
-    /*CGRect frame = self.view.frame;
-    frame.origin.y = -self.view.frame.size.height; //optional: if you want the view to drop down
-    DraggableViewBackground *draggableBackground = [[DraggableViewBackground alloc]initWithFrame:frame];
-    draggableBackground.alpha = 0; //optional: if you want the view to fade in
-
-    [self.view addSubview:draggableBackground];
-
-      //optional: animate down and in
-    [UIView animateWithDuration:0.5 animations:^{
-        draggableBackground.center = self.view.center;
-        draggableBackground.alpha = 1;
-    }];
-    */
+    [self showCards];
 }
 
+-(void)showCards{
+    self.draggableBackground = [[DraggableViewBackground alloc]initWithFrame:self.view.frame];
+    self.draggableBackground.preferences = self.preferences;
+    [self.draggableBackground fetchRecipes];
+    
+    [self.view addSubview:self.draggableBackground];
+}
 
+-(void)sendData:(NSString *)prefRequest{
+    if(prefRequest != (id)[NSNull null] && prefRequest.length != 0){
+        self.preferences = prefRequest;
+        [self.draggableBackground removeFromSuperview];
+        [self viewDidLoad];
+    }
+}
```

Comment:
What's that? 
also, not sure why you are calling viewDidLoad from here. usually viewDidLoad happens only once, and if you need to call any setup methods, you can just call it from both viewDidLoad and from another method after pulling data for example